src/file.cpp

```diff
@@ -91,36 +95,106 @@ file file::copy(const fs::path& path, std::string_view label,
 }
 
 std::string file::read() const {
-  try {
+#ifdef _WIN32
+  if (!binary) {
     std::ifstream stream = input_stream();
-    stream.exceptions(std::ifstream::failbit | std::ifstream::badbit);
-
-    return std::string(std::istreambuf_iterator(stream), {});
-  } catch (const std::ios::failure& err) {
-    throw fs::filesystem_error("Cannot read a temporary file", err.code());
+    return std::string(std::istreambuf_iterator<char>(stream), {});
   }
+#endif
+
+  native_handle_type handle = native_handle();
+
+#ifdef _WIN32
+  SetFilePointer(handle, 0, nullptr, FILE_BEGIN);
+#else
+  lseek(handle, 0, SEEK_SET);
+#endif
+
+  std::string content;
+  content.resize(file_size(path()));
```

Comment:
The file size can change during reading

While this is a undefined behavior, better not to crash here